### ISSUE TITLE
Warnings should not be rendered in red

### ISSUE THREAD

**Issue by jhjourdan** [psteckler/ProofGeneral#95]
Consider the following Coq code fragment:

```
Structure S := { x : nat -> nat }.
Canonical Structure C := {| x y := y + 1 |}.
Lemma L : True. Proof. trivial. Qed.
```

The second instructions triggers a warning.

When typing C-c b, the four first characters are rendered in red:

![screenshot](https://user-images.githubusercontent.com/1917144/32838304-bcedc4e0-ca10-11e7-9adc-9f5d1a3b20c6.png)

 There are therefore two issues:
1- why only the first four characters?
2- The rendering should be either yellow (i.e., like when proceeding step-by-step) or blue

**Comment by psteckler** [comment]
Here's the traffic when the second line is submitted:
```
*emacs:
<call val="Add"><pair><pair><string>Canonical&nbsp;Structure&nbsp;C&nbsp;:=&nbsp;{|&nbsp;x&nbsp;y&nbsp;:=&nbsp;y&nbsp;+&nbsp;1&nbsp;|}.</string>
<int>-3</int>
</pair>
<pair><state_id val="3"/><bool val="true"/></pair>
</pair>
</call>

*coqtop:
<value val="good"><pair><state_id val="4"/><pair><union val="in_l"><unit/></union><string></string></pair></pair></value>
```
So the state id corresponding to that statement is 4. Later, we see a feedback:
```
*coqtop:
<feedback object="state" route="0"><state_id val="4"/><feedback_content val="message"><message><message_level val="warning"/><option val="some"><loc start="0" stop="4"/></option><richpp><_><pp>Projection&nbsp;value&nbsp;has&nbsp;no&nbsp;head&nbsp;constant:&nbsp;<constr.keyword>fun</constr.keyword>&nbsp;y&nbsp;:&nbsp;<constr.reference>nat</constr.reference>&nbsp;=&gt;&nbsp;<constr.variable>y</constr.variable><constr.notation>&nbsp;+</constr.notation>&nbsp;1&nbsp;in&nbsp;canonical
instance&nbsp;C&nbsp;of&nbsp;x,&nbsp;ignoring&nbsp;it.&nbsp;[projection-no-head-constant,typechecker]</pp></_></richpp></message></feedback_content></feedback>
```
Note the `loc` field; it's Coq saying that the first 4 characters of that statement have something wrong, unless I've missed something.

I can change the color for warnings to yellow, and I guess those should appear in the status bar at the top.

**Comment by psteckler** [comment]
I've committed code that marks warnings in yellow and that yellow shows up in the navigation bar at top.

Note that PG/xml uses the emacs next-error protocol, and I believe that cycling among the errors includes warnings (unless there's a request to change that).

**Comment by jhjourdan** [comment]
This fixes the issue, thanks!

However, there is still something strange: The navigation bar does not seem to be affected, and, moreover, the yellow color disappears after a few seconds. Is that intended? Is there a way to deactivate this behavior?

**Comment by jhjourdan** [comment]
See [https://github.com/coq/coq/issues/6172](url) for the related Coq bug.

**Comment by psteckler** [comment]
Which version of Coq are you using? With `master`, which should be close to 8.7, the navigation bar shows the yellow, and the color persists.

**Comment by jhjourdan** [comment]
I'm using Coq 8.7.dev. The problem exists even if my .emacs file contains only these two instructions:

```
(add-to-list 'load-path "~/.emacs.d/lisp/")
(require 'proof-site "~/.emacs.d/lisp/PG/generic/proof-site")
```

**Comment by jhjourdan** [comment]
Rectification: the yellow color disappears if the commands of the script I gave are entered one after the other by using C-c C-n.

After a few seconds, the yellow colors either becomes darker blue or light blue. In any case, the color is not shown in the status bar.

**Comment by psteckler** [comment]
Yes, the temporary yellow color happens when you run the script line-by-line. That also what happens when you get an error. Is that behavior OK?

If you run the whole buffer with `C-c C-b`, the yellow color should persist (although it's only the first 4 characters, as you noted), and there should be a yellow block in the navigation bar.

**Comment by jhjourdan** [comment]
That's indeed the behavior I am seeing.

However, I am not sure I understand the rationale behind this. To me, the behavior of C-c C-b should be as close as possible to stepping through the file.

**Comment by psteckler** [comment]
If you're stepping line-by-line, you're paying attention to each statement, so the temporary color should give you notice of the warning.

If you're running the whole buffer, which may take a lot of time, you may not notice warnings as they appear. So the persistent color and navigation bar block allow you to notice them.

**Comment by jhjourdan** [comment]
> If you're stepping line-by-line, you're paying  attention to each statement, so the temporary color should give you  notice of the warning.

I also sometimes stepping line-by-line a large portion of my file (in practice by pressing F5 which is bound to C-c C-n in my emacs), so I do not pay much attention of what's happening.

You're probably going to say that this is not the right way to do it (and, arguably, I might agree with you), but I do not see why making these colors persistent might be an issue.

This can also be used as a check once the whole file is processed, to make sure that there is no warning left in the file.

**Comment by psteckler** [comment]
OK, it's reasonable to ask for persistence of the warnings. 

Can I ask that we defer making a change until the software is released, and others can offer opinions?

**Comment by jhjourdan** [comment]
Sure.

**Comment by jhjourdan** [comment]
On a related note, I do not see why errors are temporarily shown in yellow and then in red. But let's wait for more feedback.

**Comment by psteckler** [comment]
The temporary yellow color is to alert you as you go line-by-line. The red color is to mark it as an error  (as opposed to a warning). Note that there's hover-text associated with the red color, in case the error message is no longer in the \*response\* buffer.

Yes, let's see how others like it.

**Comment by jhjourdan** [comment]
> Note that there's hover-text associated with the red color, in case the error message is no longer in the *response* buffer.

Oh, nice! I've never noticed that! :)

**Comment by psteckler** [comment]
The warnings with (now) persistent yellow coloring also have hover-text.
